pkg/storage/inmemory/article.go

```diff
@@ -0,0 +1,97 @@
+package inmemory
+
+import (
+	"errors"
+	"github.com/reiver/go-porterstemmer"
+	"news-aggregator/pkg/model"
+	"strings"
+	"time"
+)
+
+// ArticleInMemory is a struct that contains the in-memory database for articles.
+type ArticleInMemory struct {
+	Articles []model.Article
+	nextID   int
+}
+
+func New() *ArticleInMemory {
+	return &ArticleInMemory{
+		Articles: []model.Article{},
+		nextID:   1,
+	}
+}
+
+// GetAll returns all articles in the database.
+func (a *ArticleInMemory) GetAll() ([]model.Article, error) {
+	return a.Articles, nil
+}
+
+// Create adds a new article to the database.
+func (a *ArticleInMemory) Create(article model.Article) (model.Article, error) {
+	article.Id = a.nextID
```

Comment:
what if there the data already exists in the db?

Reply:
In my implementation we can create article with same titles (or other fields), but it will have new ID, because higher level haven't got access for setting IDs for saving, so it will receive just new ID in db